_posts/2018-01-25-totw-10.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "Tip of the Week #10: Splitting Strings, not Hairs"
+layout: tips
+sidenav: side-nav-tips.html
+published: false
+permalink: tips/10
+type: markdown
+order: "010"
+---
+
+Originally published as totw/10 on 2012-08-16
+
+*By Greg Miller [(jgm@google.com)](mailto:jgm@google.com)*
+
+Updated 2018-01-24
+
+*I tend to have an odd split in my mind. --John Cleese*
+
+Splitting a string into substrings is a common task in any general-purpose
+programming language, and C++ is no exception. When the need arose at Google,
+many engineers found themselves wading through a morass of splitting functions
+in organically grown header files. You’d have hunted for the magical combination
+of input parameters, output parameters, and semantics that satisfies your need.
+After studying 50+ functions in a 600+ line header, you may have finally decided
+on something as tortuously named as
+`SplitStringViewToDequeOfStringAllowEmpty()`.
+
+To address this, the C++ Library Team implemented a new API for splitting
+strings, available for use in
+[absl/strings/str_split.h][str_split]
+
+The new API replaced many of the pandora’s box of splitting functions with a
```

Comment:
drop "pandora's box of"?  (weird phrasing at minimum)

Reply:
Done